src/csharp/Sequences.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class Sequences : IDisposable
+    {
+        private IntPtr _sequencesHandle;
+        private bool _disposed = false;
+        private ulong _numSequences;
+
+        internal Sequences(IntPtr sequencesHandle)
+        {
+            _sequencesHandle = sequencesHandle;
+            _numSequences = NativeMethods.OgaSequencesCount(_sequencesHandle).ToUInt64();
+        }
+
+        internal IntPtr Handle { get { return _sequencesHandle; } }
+
+        public ulong NumSequences { get { return _numSequences; } }
+
+        public ReadOnlySpan<int> this[ulong sequenceIndex]
+        {
+            get
+            {
+                ulong sequenceLength = NativeMethods.OgaSequencesGetSequenceCount(_sequencesHandle, (UIntPtr)sequenceIndex).ToUInt64();
+                IntPtr sequencePtr = NativeMethods.OgaSequencesGetSequenceData(_sequencesHandle, (UIntPtr)sequenceIndex);
+                unsafe
+                {
+                    return new ReadOnlySpan<int>(sequencePtr.ToPointer(), (int)sequenceLength);
+                }
+            }
+        }
+
+        public int[][] ToArray()
```

Comment:
I would remove this method. Clients can call them if they wanted, but that would be their code generating garbage